src/omnicore/omnicore.cpp

```diff
@@ -4119,6 +4114,11 @@ int mastercore_handler_block_end(int nBlockNow, CBlockIndex const * pBlockIndex,
         const std::string& msg = strprintf("Shutting down due to failed checkpoint for block %d (hash %s)\n", nBlockNow, pBlockIndex->GetBlockHash().GetHex());
         PrintToLog(msg);
         if (!GetBoolArg("-overrideforcedshutdown", false)) AbortNode(msg, msg);
+    } else {
```

Comment:
Well yeah, this is sort of what I'm wondering about:

- case 1: checkpoint is invalid, shutdown (thus no state changed)
- case 2: checkpoint is invalid, but user explicitly requests not to shutdown

In such a case, persisting a "wrong" state may be desired. This isn't useful, but I also don't see how it would be worse.